src/main/java/net/imglib2/algorithm/hough/HoughTransforms.java

```diff
@@ -0,0 +1,343 @@
+	{
+		voteLines( input, votespace, nRho, nTheta, Util.getTypeFromInterval( input ).createVariable() );
+	}
+
+	/**
+	 * 
+	 * Runs a Hough Line Tranform on an image and populates the vote space
+	 * parameter with the results.
+	 * <p>
+	 * Vote space here has two dimensions: {@code rho} and {@code theta}.
+	 * {@code theta} is measured in radians {@code [-pi/2 pi/2)}, {@code rho} is
+	 * measured in {@code [-rhoMax, rhoMax)}.
+	 * </p>
+	 * <p>
+	 * Lines are modeled as
+	 * </p>
+	 * 
+	 * <pre>
+	 * l(t) = | x | = rho * |  cos(theta) | + t * | sin(theta) |
+	 *        | y |         | -sin(theta) |       | cos(theta) |
+	 * </pre>
+	 * <p>
+	 * In other words, {@code rho} represents the signed minimum distance from
+	 * the image origin to the line, and {@code theta} indicates the angle
+	 * between the row-axis and the minimum offset vector.
+	 * </p>
+	 * <p>
+	 * For a given point, then, votes are placed along the curve
+	 * </p>
+	 * 
+	 * <pre>
+	 * rho = y * sin( theta ) + x * cos( theta )
+	 * </pre>
+	 * <p>
+	 * It is important to note that the interval of the first dimension of the
+	 * vote space image is NOT {@code [-maxRho, maxRho)} but instead
+	 * {@code [0, maxRho * 2)}. Thus if {@link HoughTransforms#pickPeaks} is not
+	 * used to retrieve the maxima from the vote space, the vote space will have
+	 * to be translated by {@code -maxRho} in dimension 0 to get the correct
+	 * {@code rho} and {@code theta} values from the vote space.
+	 * </p>
+	 * 
+	 * @param input
+	 *            - the {@link RandomAccessibleInterval} to run the Hough Line
+	 *            Transform over
+	 * @param votespace
+	 *            - the {@link RandomAccessibleInterval} in which the results
+	 *            are stored
+	 * @param nRho
+	 *            - the number of bins for rho resolution
+	 * @param nTheta
+	 *            - the number of bins for theta resolution
+	 * @param threshold
+	 *            - the minimum value allowed by the populator. Any input less
+	 *            than or equal to this value will be disregarded by the
+	 *            populator.
+	 */
+	public static < T extends RealType< T > > void voteLines( final RandomAccessibleInterval< T > input, final RandomAccessibleInterval< T > votespace, final int nRho, final int nTheta, final T threshold )
+	{
+
+		final long[] dims = new long[ input.numDimensions() ];
+		input.dimensions( dims );
+
+		final double minRho = -computeLength( dims );
+		final double dRho = 2 * computeLength( dims ) / ( double ) nRho;
+
+		final double minTheta = -Math.PI / 2;
+		final double dTheta = Math.PI / nTheta;
+
+		final double[] rho = new double[ nRho ];
+		final double[] theta = new double[ nTheta ];
+
+		// create theta LUT
+		for ( int t = 0; t < nTheta; ++t )
+		{
+			theta[ t ] = dTheta * t + minTheta;
+		}
+
+		// create rho LUT
+		for ( int r = 0; r < nRho; ++r )
+		{
+			rho[ r ] = dRho * r + minRho;
+		}
+
+		final Cursor< T > imageCursor = Views.iterable( input ).localizingCursor();
```

Comment:
For the time being I think this is the way to go. I am still wondering if it is possible to input an image with `min != [0, ... , 0]`, but all of my attempts so far to get results from such an image have failed. I may attempt to solve this in the end but for now I will do this. My only concern was that if the user passes through a `RAI` with `min != [0, ... , 0]` the resulting `(rho, theta)` coordinates would not be technically correct (since we are translating the image we are translating the `rho` of the lines). I may attempt to fix this but with the current algorithm it may be a big time commitment